programs/mango-v4/src/instructions/flash_loan.rs

```diff
@@ -391,18 +389,31 @@ pub fn flash_loan_end<'key, 'accounts, 'remaining, 'info>(
             approved_amount
         };
 
+        let loan_origination_fee = cm!(loan * bank.loan_origination_fee_rate);
+        cm!(bank.collected_fees_native += loan_origination_fee);
+
+        if bank.is_reduce_only() {
+            let change_amount = if change.amount >= 0 {
+                cm!(change.amount - loan_origination_fee)
+            } else {
+                cm!(change.amount + loan_origination_fee)
+            };
+            require!(
+                (change_amount < 0 && native + change_amount >= 0)
+                    || (change_amount > 0 && native + change_amount < 2),
+                MangoError::SomeError
+            );
+        }
+
         // Enforce min vault to deposits ratio
-        if loan > 0 {
+        if native + change.amount - loan_origination_fee < 0 {
```

Comment:
cm, and since we also need it inside the `if reduce-only`, maybe pre-compute it before that block